pkg/validation/cluster.go

```diff
@@ -0,0 +1,129 @@
+package validation
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubelet/config/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/k8stopologyawareschedwg/deployer/pkg/validator"
+
+	mcov1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+
+	"github.com/openshift-kni/numaresources-operator/pkg/kubeletconfig"
+)
+
+func ValidateCluster(aContext context.Context, aClient client.Client) ([]validator.ValidationResult, error) {
```

Comment:
we can have multiple `kubeletconfig` and we need to check that either

- the final rendered kubelet has the values we need (best)
- at least a kubelet snippet has the values we need (weaker guarantee, but we can start from here)